### PR TITLE
fix: Keeping no columns with Table.keep_only_columns results in an empty Table with a row count above 0

### LINKED ISSUE #318: Keeping no columns with `Table.keep_only_columns` results in an empty Table with a row count above 0
### Describe the bug

If you use the method `Table.keep_only_columns` with an empty list on a table that contains at least one row, you get an empty table with a row count above 0.

The methods `Table.remove_columns` has the same bug if you remove all columns.

### To Reproduce

```py
from safeds.data.tabular.containers import Table

table = Table({"Column1": [1, 2, 3], "Column2": [4, 5, 6]})
table = table.keep_only_columns([])

print(table)
print(table.number_of_rows)
```

The print statement returns 3 but the table is empty

### Expected behavior

The row count should be 0 if the table is empty

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by alex-senger**
Closes #318 

Fixes Bug:

If you use the method Table.keep_only_columns with an empty list on a table that contains at least one row, you get an empty table with a row count above 0.

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/7.0.4) status: ✅ [SUCCESS](https://github.com/Safe-DS/Stdlib/actions/runs/5356473478)

| Descriptor  |                                Linter                                 |Files|Fixed|Errors|Elapsed time|
|-------------|-----------------------------------------------------------------------|-----|----:|------|------------|
|✅ PYTHON    |[black](https://megalinter.io/7.0.4/descriptors/python_black)          |    3|    1|     0|1.0s        |
|✅ PYTHON    |[mypy](https://megalinter.io/7.0.4/descriptors/python_mypy)            |    3|     |     0|2.26s       |
|✅ PYTHON    |[ruff](https://megalinter.io/7.0.4/descriptors/python_ruff)            |    3|    1|     0|0.03s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/7.0.4/descriptors/repository_git_diff)|yes  |     |no    |0.0s        |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5356473478)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/tabular/containers/_table.py [hunk 1]
```diff
@@ -355,7 +356,7 @@ def __eq__(self, other: Any) -> bool:
             return NotImplemented
         if self is other:
             return True
-        if self.number_of_rows == 0 and other.number_of_rows == 0:
+        if self.number_of_columns == 0 and other.number_of_columns == 0:
             return self.column_names == other.column_names
```
Comment (sibre28) [inline src/safeds/data/tabular/containers/_table.py:360]:
```suggestion
            return true
```
If neither Table has any columns, they are both empty tables and therefore always equals

**Review by sibre28 — CHANGES_REQUESTED** [review]

**Review by sibre28 — APPROVED** [review]
[LGTM](https://lgtm.party/p/GSQE7HO/)

**Pull request merged**